projects/map-ol/src/lib/map-ol.service.ts

```diff
@@ -812,11 +892,106 @@ export class MapOlService {
     return newlayer;
   }
 
-  private create_wms_layer(l: WmsLayer): olTileLayer<olTileSource> {
+  private create_tms_layer(l: RasterLayer | VectorLayer): olTileLayer<olTileSource> {
+    let newlayer = null;
+    let newOlSource = null;
+    if (l instanceof RasterLayer) {
+      newlayer = this.create_xyz_layer(l);
+      newlayer.set('type', 'tms');
+
+    } else if (l instanceof VectorLayer) {
+      newOlSource = new olVectorTileSource({
+        format: new olMVT(),
+        tileGrid: createXYZ({ minZoom: l.minZoom || undefined, maxZoom: l.maxZoom || undefined }),
+        url: l.url,
+        wrapX: false
+      });
+
+
+      if (l.attribution) {
+        newOlSource.setAttributions([l.attribution]);
+      }
+
+      if (l.continuousWorld) {
+        newOlSource.set('wrapX', l.continuousWorld);
+      }
+
+      if (l.subdomains) {
+        const urls = l.subdomains.map((item) => l.url.replace('{s}', `${item}`));
+        newOlSource.setUrls(urls);
+
+      } else {
+        newOlSource.setUrl(l.url);
+      }
+
+
+      const layeroptions: any = {
+        type: 'tms',
+        filtertype: l.filtertype,
+        name: l.name,
+        id: l.id,
+        visible: l.visible,
+        legendImg: l.legendImg,
+        opacity: l.opacity || 1,
+        zIndex: 1,
+        source: newOlSource,
+        declutter: true,
+        renderMode: 'hybrid'
+      };
+
+      if (l.popup) {
+        layeroptions.popup = l.popup;
+        /**
+         * ol 6.x problem if popup (map.forEachLayerAtPixel) use className
+         * https://github.com/openlayers/openlayers/releases/tag/v6.0.0
+         */
+        layeroptions.className = l.id;
+      }
+
+      if (l.bbox) {
+        layeroptions.extent = transformExtent(l.bbox.slice(0, 4) as [number, number, number, number], WGS84, this.getProjection().getCode());
+      }
+
+      if (l.maxResolution) {
+        layeroptions.maxResolution = l.maxResolution;
+      }
+      if (l.minResolution) {
+        layeroptions.minResolution = l.minResolution;
+      }
+
+      if (l.maxZoom) {
+        layeroptions.maxZoom = l.maxZoom;
+      }
+      if (l.minZoom) {
+        layeroptions.minZoom = l.minZoom;
+      }
+
+      newlayer = new olVectorTileLayer(layeroptions);
+      this.setCrossOrigin(l, newlayer);
+      this.addEventsToLayer(l, newlayer, newOlSource);
+
+
+      const style = l?.options?.style;
+      const mapboxSourceKey = l?.options?.styleSource;
+      if (style && mapboxSourceKey) {
+        applyStyle(newlayer, style, mapboxSourceKey);
+      }
+    }
+    return newlayer;
+  }
+
+  private create_wms_layer(l: WmsLayer) {
+    let newlayer: olTileLayer<olTileSource> | olImageLayer<olImageSource> = this.create_tiled_wms_layer(l);
```

Comment:
Maybe here it would be better to create the layer in an if clause, so in case TILED === 'false' we dont have to run both functions.